File: 天梯赛/java/L1-020. 帅到没朋友.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args){
		Scanner in=new Scanner(System.in);
		int n=in.nextInt();
		boolean mark[]=new boolean [100000];
		for(int i=0;i<n;++i) {
			int k=in.nextInt();
			for(int j=0;j<k;++j) {
				int tmp=in.nextInt();
				if(k!=1)mark[tmp]=true;
			}
		}
		int m=in.nextInt();
		boolean first=true;
		for(int i=0;i<m;++i) {
			int tmp=in.nextInt();
			if(!mark[tmp]) {
				if(!first)System.out.print(" ");
				first=false;
				System.out.printf("%05d",tmp);
				mark[tmp]=true;
			}
		}
		in.close();
		if(first)System.out.print("No one is handsome");
	}	
	
}

File: 天梯赛/L1-020. 帅到没朋友.cpp
#include<iostream>
#include<cstring>
#include<vector>
#include<string>
#include<cstring>
#include<queue>
#include<set>
#include<stack>
#include<map>
#include<cmath>
#include<cstdio>
#include<algorithm>
using namespace std;
#define lowbit(i) ((i)&(-i))
typedef long long ll;
const int maxn=1e5+10;
const int inf=0x3f3f3f3f;
map<int,bool> m;
int main(){
	//ios::sync_with_stdio(false);
	int n;
	cin>>n;
	while(n--){
		int k,tmp;
		cin>>k;
		if(k==1)cin>>tmp;
		else
			for(int j=0;j<k;++j){
				cin>>tmp;
				m[tmp]=true;
			}
	}
	int q;
	cin>>q;
	bool first=true;
	while(q--){
		int tmp;
		cin>>tmp;
		if(!m[tmp]){
			if(!first)cout<<" ";
			first=false;
			printf("%05d",tmp);
		}
	}
	if(first){
		cout<<"No one is handsome";
	}
	return 0;
}
